Microsoft.Toolkit.Uwp.Services/Services/MicrosoftTranslator/AzureAuthToken.cs

```diff
@@ -0,0 +1,133 @@
+                {
+                    // If the subscription key is changed, the token is no longer valid.
+                    _subscriptionKey = value;
+                    _storedTokenValue = string.Empty;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureAuthToken"/> class, that is used to obtain access token
+        /// </summary>
+        /// <param name="key">Subscription key to use to get an authentication token.</param>
+        public AzureAuthToken(string key)
+        {
+            SubscriptionKey = key;
+            _client = new HttpClient();
+        }
+
+        /// <summary>
+        /// Gets a token for the specified subscription.
+        /// </summary>
+        /// <returns>The encoded JWT token prefixed with the string "Bearer ".</returns>
+        /// <remarks>
+        /// This method uses a cache to limit the number of request to the token service.
+        /// A fresh token can be re-used during its lifetime of 10 minutes. After a successful
+        /// request to the token service, this method caches the access token. Subsequent
+        /// invocations of the method return the cached token for the next 5 minutes. After
+        /// 5 minutes, a new token is fetched from the token service and the cache is updated.
+        /// </remarks>
+        public async Task<string> GetAccessTokenAsync()
+        {
+            if (string.IsNullOrEmpty(SubscriptionKey))
+            {
+                throw new ArgumentNullException(nameof(SubscriptionKey), "A subscription key is required");
+            }
+
+            // Re-use the cached token if there is one.
+            if ((DateTime.Now - _storedTokenTime) < TokenCacheDuration && !string.IsNullOrWhiteSpace(_storedTokenValue))
+            {
+                return _storedTokenValue;
+            }
+
+            using (var request = new HttpRequestMessage())
+            {
+                request.Method = HttpMethod.Post;
+                request.RequestUri = ServiceUrl;
+                request.Content = new StringContent(string.Empty);
+                request.Headers.TryAddWithoutValidation(OcpApimSubscriptionKeyHeader, this.SubscriptionKey);
+
+                var response = await _client.SendAsync(request).ConfigureAwait(false);
+                response.EnsureSuccessStatusCode();
+
+                var token = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                _storedTokenTime = DateTime.Now;
+                _storedTokenValue = $"Bearer {token}";
+
+                return _storedTokenValue;
+            }
+        }
+
+        /// <summary>
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+        /// </summary>
+        public void Dispose()
+        {
+            try
+            {
+                _client.Dispose();
+            }
+            catch
```

Comment:
is there is a case for using try catch at the client dispose ?? I see it is redundant also if you used the httpHelper you will not have to worry about managing the httpClient.